### ISSUE TITLE
Index is lost after using SET clause

### ISSUE THREAD

**Issue by pdpotter** [apache/incubator-age#161]
**Describe the bug**
After updating data using the SET clause, data from the updated vertex is no longer in the index.

**How are you accessing AGE (Command line, driver, etc.)?**
- psql

**What data setup do we need to do?**
```pgsql
...
CREATE OR REPLACE FUNCTION get_id(properties agtype) 
  RETURNS agtype
AS
$BODY$
    select agtype_access_operator($1, '"id"');
$BODY$
LANGUAGE sql
IMMUTABLE;
CREATE FUNCTION

SELECT * FROM cypher('test_graph', $$CREATE (p:person {id: 1, name: 'Jane'}) return p$$) as (p agtype);
p
---------------------------------------------------------------------------------------------
 {"id": 844424930131971, "label": "person", "properties": {"id": 1, "name": "Jane"}}::vertex
(1 row)

CREATE UNIQUE INDEX person_id_idx ON test_graph.person(get_id(properties));
CREATE INDEX

SELECT * FROM cypher('test_graph', $$Match (p:person {id: 1}) set p.name = "Janine"  return p$$) as (p agtype);
p
---------------------------------------------------------------------------------------------
 {"id": 844424930131971, "label": "person", "properties": {"id": 1, "name": "Janine"}}::vertex
(1 row)
...
```

**What is the necessary configuration info needed?**
/

**What is the command that caused the error?**
```pgsql
SELECT * FROM cypher('test_graph', $$CREATE (p:person {id: 1, name: 'Jane'}) return p$$) as (p agtype);                                              p                                              
---------------------------------------------------------------------------------------------
 {"id": 844424930131973, "label": "person", "properties": {"id": 1, "name": "Jane"}}::vertex
(1 row)
```

**Expected behavior**
I would expect an error message `ERROR:  duplicate key value violates unique constraint "person_id_idx"
DETAIL:  Key (get_id(properties))=(1) already exists.`. Instead, the new vertex is added, leading to two vertices with id 1:

```pgsql
SELECT * FROM cypher('test_graph', $$Match (p:person {id: 1})  return p$$) as (p agtype);
                                               p                                               
-----------------------------------------------------------------------------------------------
 {"id": 844424930131971, "label": "person", "properties": {"id": 1, "name": "Janine"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"id": 1, "name": "Jane"}}::vertex
(2 rows)
```

This should be impossible because of the unique index that has been created.

**Environment (please complete the following information):**
- Version: bf9d8367bf4f9ba03d8b79aee5e2065c7a9edbc8

**Additional context**
I encountered this bug while debugging an issue with a corrupt index on the ids of vertices, which was caused by a query with an invalid syntax (see https://github.com/apache/incubator-age/issues/160).

**Comment by bravius** [comment]
Hey bud, I was also looking for an indexing solution, so I just re-worked your function as a plpgsql function. It seems to work as expected now, throws error on duplicates. Thus, I don't think it was a bug, just the way the custom function was written:

```
CREATE OR REPLACE FUNCTION get_id(properties agtype)
RETURNS agtype
AS
$BODY$
BEGIN
   return agtype_access_operator(properties, '"id"');
END;
$BODY$
LANGUAGE plpgsql
IMMUTABLE;
```

**Comment by JoshInnis** [comment]
@pdpotter Have you confirmed @bravius 's finidings, or is this still an issue?

**Comment by bravius** [comment]
@pdpotter 

Okay I just checked that.

When CREATE-ing a vertex with a duplicate key, I get "duplicate key value violates unique constraint", as expected.

When I use MATCH and SET to update an existing vertex with a duplicate key, no error is thrown and the index seems to be completely ignored.

It seems that SET syntax is bypassing index constraint checking, but CREATE syntax does get checked.

**Comment by bravius** [comment]
@JoshInnis - It's still an issue. CREATE throws errors on index duplicates, but SET does not.

**Comment by JoshInnis** [comment]
This issue should now be resolved in master and should be included in the next (1.1.0) release

**Comment by pdpotter** [comment]
The issue is indeed resolved in master. Thank you!
